pkg/transparentproxy/validate/validator.go

```diff
@@ -0,0 +1,191 @@
+package validate
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"math/big"
+	"net"
+	"net/netip"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/sethvargo/go-retry"
+)
+
+const (
+	ValidationServerPort uint16 = 15006
+	validationRetries           = 10
+	validationInterval          = 1 * time.Second
+)
+
+type Validator struct {
+	Config *Config
+	Logger logr.Logger
+}
+
+type Config struct {
+	ServerListenIP      netip.Addr
+	ServerListenPort    uint16
+	ClientConnectIP     netip.Addr
+	ClientConnectPort   uint16
+	ClientRetryInterval time.Duration
+}
+
+func NewValidator(useIpv6 bool, port uint16, logger logr.Logger) *Validator {
+	// Traffic to lo (but not 127.0.0.1) by sidecar will be redirected to  KUMA_MESH_INBOUND_REDIRECT, so:
+	// connect to 127.0.0.6 should be redirected to 127.0.0.1
+	// connect to ::6       should be redirected to ::1
+	serverListenIP, _ := netip.AddrFromSlice([]byte{127, 0, 0, 1})
+	clientConnectIP, _ := netip.AddrFromSlice([]byte{127, 0, 0, 6})
+
+	if useIpv6 {
+		serverListenIP, _ = netip.AddrFromSlice([]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1})
+		clientConnectIP, _ = netip.AddrFromSlice([]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 6})
+	}
+
+	return &Validator{
+		Config: &Config{
+			ServerListenIP:      serverListenIP,
+			ServerListenPort:    port,
+			ClientConnectIP:     clientConnectIP,
+			ClientRetryInterval: validationInterval,
+		},
+		Logger: logger,
+	}
+}
+
+func (validator *Validator) Run() error {
+	validator.Logger.Info("Starting iptables validation...")
```

Comment:
```suggestion
	validator.Logger.Info("starting iptables validation")
```

Don't use elipsis and start lowercase - check out our logging madr https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/003-logging.md?plain=1#L122-L130 https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/003-logging.md?plain=1#L34